pkg/ipcache/ipcache.go

```diff
@@ -267,10 +318,51 @@ func (ipc *IPCache) Upsert(ip string, hostIP net.IP, hostKey uint8, k8sMeta *K8s
 		ipc.ipToHostIPCache[ip] = IPKeyPair{IP: hostIP, Key: hostKey}
 	}
 
-	if k8sMeta == nil {
-		delete(ipc.ipToK8sMetadata, ip)
-	} else {
-		ipc.ipToK8sMetadata[ip] = *k8sMeta
+	if !metaEqual {
+		// Keep old named ports if new identity is from kv-store and has no named ports.
+		// This needs to be done to preserve the named ports as kv-store does not have them.
```

Comment:
Can't we add it to the kvstore?

Reply:
You tell me. I recall facing a comment that discouraged me from adding it.

Reply:
```
// IPIdentityPair is a pairing of an IP and the security identity to which that
// IP corresponds. May include an optional Mask which, if present, denotes that
// the IP represents a CIDR with the specified Mask.
//
// WARNING - STABLE API
// This structure is written as JSON to the key-value store. Do NOT modify this
// structure in ways which are not JSON forward compatible.
type IPIdentityPair struct {
	IP           net.IP          `json:"IP"`
	Mask         net.IPMask      `json:"Mask"`
	HostIP       net.IP          `json:"HostIP"`
	ID           NumericIdentity `json:"ID"`
	Key          uint8           `json:"Key"`
	Metadata     string          `json:"Metadata"`
	K8sNamespace string          `json:"K8sNamespace,omitempty"`
	K8sPodName   string          `json:"K8sPodName,omitempty"`
}
```

Reply:
Maybe I did not spend enough time figuring out what it takes to be "JSON forward compatible"?

Reply:
You can add fields but you can't rename or remove existing ones.

Reply:
So the consensus is to add the named ports to kv-store as well?

Reply:
Done.